service_discovery/proto/chariott/v1/chariott_registry.proto

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+syntax = "proto3";
+package chariott_registry;
+
+// Service Registry definition
+//
+// The protobuf definitions for Chariott's service registry
+
+// The entry point for the Registry gRPC Service.
+service Registry {
+  // Register, or add a service to the registry
+  rpc Register(RegisterRequest) returns (RegisterResponse) {}
+
+  // Unregister, or remove a service from the registry
+  rpc Unregister(UnregisterRequest) returns (UnregisterResponse) {}
+
+  // Discover, or retrieve the metadata for a single service given its fully qualified name
+  rpc DiscoverService(DiscoverServiceRequest) returns (DiscoverServiceResponse) {}
+
+  // Discover a list of services given their namespace
+  rpc DiscoverByNamespace(DiscoverByNamespaceRequest) returns (DiscoverByNamespaceResponse) {}
+
+  // Inspect, or retrieve all contents of the service registry
+  rpc Inspect(InspectRequest) returns (InspectResponse) {}
+}
+
+// Represenation of a service including all of its metadata that the registry stores
+message ServiceMetadata {
+  string namespace = 1;
+  string name = 2;
+  string version = 3;
+  string uri = 4;
+  string communication_kind = 5;
+  string communication_reference = 6;
```

Comment:
can we add some samples for what the two communication_* fields are?

Reply:
Added comments for all of the fields in ServiceMetadata